mne/simulation/evoked.py

```diff
@@ -196,9 +198,14 @@ def _generate_noise(info, cov, iir_filter, random_state, n_samples, zi=None):
     rng = check_random_state(random_state)
     c = np.diag(noise_cov.data) if noise_cov['diag'] else noise_cov.data
     mu_channels = np.zeros(len(c))
+    stds = np.sqrt(np.diag(c))
+    stds_inv = np.zeros_like(stds)
+    stds_inv[stds != 0.] = 1. / stds[stds != 0.]
+    c = np.diag(stds_inv).dot(c).dot(np.diag(stds_inv))
     # we almost always get a positive semidefinite warning here, so squash it
     with warnings.catch_warnings(record=True):
```

Comment:
The semidefinites are causing a problem in tests, probably need a `warnings.simplefilter('ignore')` line added below this